src/phylum/ci/git.py

```diff
@@ -45,6 +46,29 @@ def git_remote(git_c_path: Optional[Path] = None) -> str:
     return remote
 
 
+def git_fetch(repo: Optional[str] = None, ref: Optional[str] = None, git_c_path: Optional[Path] = None) -> None:
+    """Execute a `git fetch` command with optional repository and refspec specified.
+
+    See git documentation for more detail: https://git-scm.com/docs/git-fetch
+
+    The optional `git_c_path` is used to tell `git` to run as if it were started in that
+    path instead of the current working directory, which is the default when not provided.
+    """
+    base_cmd = git_base_cmd(git_c_path=git_c_path)
+    cmd = [*base_cmd, "fetch"]
+    if repo is not None:
+        cmd.append(repo)
+        # Specifying a refspec is only possible when a repository is already specified
+        if ref is not None:
+            cmd.append(ref)
+    LOG.debug("Executing command: %s", shlex.join(cmd))
+    try:
+        subprocess.run(cmd, check=True, capture_output=True, text=True)  # noqa: S603
+    except subprocess.CalledProcessError as err:
+        msg = "Fetching failed. Ensure credentials are available to run git commands."
```

Comment:
This error might even be a bit too specific? We don't know why the process failed and there's a million reasons. Though I suppose credentials are the most likely culprit.

Reply:
The docstring for `PhylumCalledProcessError` specifies that `msg` is meant to "provide a best effort explanation about what went wrong and what to do about it.":

<img width="1554" alt="image" src="https://github.com/phylum-dev/phylum-ci/assets/18729796/bb195f12-50b6-4cdb-8b9f-72d61261b7b4">

The idea is that _something_ is provided in error output to give user's a starting point. As pointed out, improper credentials is the most likely reason in this situation.